src/App.js

```diff
@@ -3,64 +3,34 @@ import "./App.css";
 import CurrencyRow from "./components/CurrencyRow";
 import CurrencyConverterText from "./components/CurrencyConverterText";
 import TimeStamp from "./components/TimeStamp";
+import useCurrencyConverter from "./components/useCurrencyConverter";
 
 const endpoint = "latest";
 const access_key = process.env.REACT_APP_EXCHANGE_RATES_API_KEY;
 
-const params = {
-  access_key: access_key,
-  endpoint: endpoint,
-};
-
-const BASE_URL = `http://api.exchangeratesapi.io/v1/${params.endpoint}?access_key=${params.access_key}`;
-
 function App() {
-  const [currencyOptions, setCurrencyOptions] = useState([]);
-  const [fromCurrency, setFromCurrency] = useState();
-  const [toCurrency, setToCurrency] = useState();
-  const [exchangeRate, setExchangeRate] = useState();
-  const [amount, setAmount] = useState(1);
-  const [isAmountInFromCurrency, setAmountInFromCurrency] = useState(true);
-  const [currencies, setCurrencies] = useState({});
+  const {
+    timestamp,
+    currencyOptions,
+    fromCurrency,
+    toCurrency,
+    exchangeRate,
+    amount,
+    isAmountInFromCurrency,
+    setFromCurrency,
+    setToCurrency,
+    setAmount,
+    setAmountInFromCurrency,
+    currencies,
+  } = useCurrencyConverter(endpoint, access_key);
```

Comment:
What guided your decision to abstract everything but the endpoint and the access_key from the hook API?

Now the hook feels heavy to use. I would suggest exploring ways to make this simpler to use.

It makes almost always sense to have hooks as easy to use as possible.

e.g.
```jsx
const { exhangeRate, convertedAmount } = useCurrencyConverter({fromCurrency, toCurrency, amount})
```

Having a nice API for hooks enhances DX greatly.